src/components/Guide/Guide.js

```diff
@@ -0,0 +1,206 @@
+    transform: scale(1);
+  }
+
+  1% {
+    opacity: 0;
+    transform: scale(1.25);
+  }
+
+  100% {
+    opacity: 1;
+    transform: scale(1);
+  }
+`
+
+const leave = keyframes`
+  0% {
+    opacity: 1;
+  }
+
+  99% {
+    opacity: 0;
+  }
+
+  100% {
+    opacity: 0;
+    z-index: -1;
+  }
+`
+
+const getPointerStyles = position => {
+  switch (position) {
+    case 'top': {
+      return css`
+        transform: rotate(-90deg);
+        left: 0;
+        bottom: -29px;
+      `
+    }
+    case 'right': {
+      return css`
+        transform: rotate(0deg);
+        left: -34px;
+        bottom: 8px;
+      `
+    }
+    case 'bottom': {
+      return css`
+        transform: rotate(90deg);
+        left: 0;
+        top: -29px;
+      `
+    }
+    case 'left': {
+      return css`
+        transform: rotate(180deg);
+        right: -34px;
+        bottom: 8px;
+      `
+    }
+    default: {
+      return css``
+    }
+  }
+}
+
+const GuideContainer = styled.div`
+  --reactour-accent: ${props => props.accentColor};
+  position: fixed;
+  transition: top 0.3s, left 0.3s;
+  opacity: 0;
+  left: ${({ coordinates }) => coordinates[0]}px;
+  top: ${({ coordinates }) => coordinates[1]}px;
+  animation: ${({ isOpen }) => (isOpen ? enter : leave)}
+  animation-duration: 0.5s;
+  animation-fill-mode: forwards;
+  background-color: #fff;
+  border: 1px solid #fff;
+  padding: 24px;
+  box-shadow: 0 0.5em 3em rgba(0, 0, 0, 0.3);
+  color: inherit;
+  z-index: 1000000;
+  max-width: 320px;
+  min-width: 150px;
+  outline: 0;
+  border-radius: ${props => props.rounded}px;
+`
+
+const PointerContainer = styled.div`
+  display: flex;
+  position: absolute;
+  z-index: 1;
+  pointer-events: none;
+  ${({ position }) => getPointerStyles(position)};
+`
+
+class Guide extends React.Component {
+  getPositioning = () => {
+    const {
+      targetTop,
+      targetRight,
+      targetBottom,
+      targetLeft,
+      windowWidth,
+      windowHeight,
+      helperWidth,
+      helperHeight,
+      helperPosition,
+      padding,
+    } = this.props
+
+    const available = {
+      left: targetLeft,
+      right: windowWidth - targetRight,
+      top: targetTop,
+      bottom: windowHeight - targetBottom,
+    }
+
+    const couldPositionAt = position => {
+      return (
+        available[position] >
+        (hx.isHoriz(position)
+          ? helperWidth + padding * 2
+          : helperHeight + padding * 2)
+      )
+    }
+
+    const autoPosition = coords => {
+      const positionsOrder = hx.bestPositionOf(available)
+      for (let j = 0; j < positionsOrder.length; j++) {
+        if (couldPositionAt(positionsOrder[j])) {
+          return {
+            coordinates: coords[positionsOrder[j]],
+            position: positionsOrder[j],
+          }
+        }
+      }
+      return {
+        coordinates: coords.center,
+        position: 'center',
+      }
+    }
+
+    const hX = hx.isOutsideX(targetLeft + helperWidth, windowWidth)
```

Comment:
i know you didn't write this, but these fiddly logic makes me a bit nervous without tests backing it

Reply:
yeah...I think in the future I'll revisit this and see about making this more local to `web-client-components`.